scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+        commitIfNeeded(channel);
+    }
+
+    @Override
+    public void clear(String originator, TaskId taskid, String channel) throws InvalidChannelException, IOException {
+        getChannel(channel).clear();
+        logWithContext(taskid, channel, "cleared channel", Level.DEBUG);
+        commitIfNeeded(channel);
+    }
+
+    @Override
+    public Set<String> keySet(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).keySet();
+    }
+
+    @Override
+    public <T extends Serializable> Collection<T> values(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).values();
+    }
+
+    @Override
+    public <T extends Serializable> Set<Map.Entry<String, T>> entrySet(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).entrySet();
+    }
+
+    @Override
+    public <T extends Serializable> T getOrDefault(String originator, TaskId taskid, String channel, String key,
+            T defaultValue) throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).getOrDefault(key, defaultValue);
+    }
+
+    @Override
+    public <T extends Serializable> void forEach(String originator, TaskId taskid, String channel, String action)
+            throws CompilationException, ClosureEvaluationException, InvalidChannelException {
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            chosenChannel.forEach((BiConsumer<? super String, ? super T>) evaluateClosure(action, BiConsumer.class));
+            logWithContext(taskid, channel, "Run forEach, new content is: " + chosenChannel, Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+
+    }
+
+    @Override
+    public <T extends Serializable> void replaceAll(String originator, TaskId taskid, String channel, String function)
+            throws InvalidChannelException, CompilationException, IOException {
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            chosenChannel.replaceAll((BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(function,
+                                                                                                          BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid, channel, "Run replaceAll, new content is: " + chosenChannel, Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)